src/GoTrueClient.ts

```diff
@@ -937,9 +938,10 @@ export default class GoTrueClient {
     }
 
     try {
-      window?.addEventListener('visibilitychange', () => {
+      window?.addEventListener('visibilitychange', async () => {
         if (document.visibilityState === 'visible') {
-          this._recoverAndRefresh()
+          await this.initializePromise
+          await this._recoverAndRefresh()
```

Comment:
This is unrelated, but I've included it in this PR. If the user quickly switches tabs while the session is still initializing, it won't call `_recoverAndRefresh` prematurely.